### PR TITLE
The Good, the Bad and the Ugly

### PR DESCRIPTION
**Pull request by darko-poljak**
Hello!

I have made some refactorings. For now only obvious (ugly->bad) parts:
- simplify some APIs
- renamed enable-beta to beta (from beta branch)
- CDIST_BETA env var (from beta branch)
- extracted argparse code into separate file
- extracted functions into separate util files.

@telmich I propose to merge before (if) we continue to do some code cleanup.

### CODE REVIEW

File: cdist/argparse.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    1: logging.WARNING,
+    2: logging.INFO,
+}
+_verbosity_level = collections.defaultdict(
+    lambda: logging.DEBUG, _verbosity_level)
+
+
+def add_beta_command(cmd):
+    if cmd not in BETA_COMMANDS:
+        BETA_COMMANDS.append(cmd)
```
Comment (telmich) [inline cdist/argparse.py]: Maybe a set() might also be appropriate here?

**Comment by telmich** [comment]
Looks good to me!

File: cdist/argparse.py (hunk 1 above)
Comment (darko-poljak) [inline cdist/argparse.py]: Good point.

**Comment by darko-poljak** [comment]
@telmich Changed list -> set for beta command and args.
Also fixed some missing vars errors.
I will merge this. Of course, we have to test it thoroughly.

_"Beware of bugs in the above code; I have only proved it correct, not tried it."_
_Donald Knuth_

**Pull request merged**